rebasehelper/tests/public_api/test_specfile.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you rebase your package to the latest version
+# Copyright (C) 2013-2019 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hráček <phracek@redhat.com>
+#          Tomáš Hozza <thozza@redhat.com>
+#          Nikola Forró <nforro@redhat.com>
+#          František Nečas <fifinecas@seznam.cz>
+
+import os
+import shutil
+
+import pytest  # type: ignore
+
+from rebasehelper.specfile import SpecFile
+from rebasehelper.tests.conftest import SPEC_FILE, TEST_FILES_DIR
+
+
+@pytest.mark.public_api
+class TestSpecFile:
+    def test_constructor(self, workdir):
+        shutil.copy(os.path.join(TEST_FILES_DIR, SPEC_FILE), workdir)
+        # check that no mandatory argument was added
+        SpecFile(SPEC_FILE)
+        # check that arguments were not renamed or removed
+        SpecFile(path=SPEC_FILE, sources_location=workdir, predefined_macros=None)
+
+    def test_update_changelog(self, spec_object):
+        entry = 'test message'
+        spec_object.update_changelog(changelog_entry=entry)
+        assert entry in spec_object.spec_content.section('%changelog')[1]
+        spec_object.update_changelog(entry)
+        assert entry in spec_object.spec_content.section('%changelog')[1]
+
+    def test_set_version(self, spec_object):
+        NEW_VERSION = '1.2.3.4.5'
+        spec_object.set_version(version=NEW_VERSION)
+        assert spec_object.header.version == NEW_VERSION
+
+        NEW_VERSION = '1.2.3.4'
+        spec_object.set_version(NEW_VERSION)
+        assert spec_object.header.version == NEW_VERSION
```

Comment:
Why the inconsistencies? Obviously version has to be different so the change can be verified, but the same applies to changelog entry. If the second call to `update_changelog()` doesn't generate a new entry at all, the second assert wouldn't catch it.